src/components/canvas.tsx

```diff
@@ -0,0 +1,54 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import { Furniture } from '../types/furniture'
+import * as React from 'react'
+import CanvasCard from '../utils/canvas'
+import debounce from 'debounce'
+
+interface CanvasProps {
+  update: (blob : Blob) => void
+  furniture?: Furniture
+}
+
+interface CanvasState {
+  card: CanvasCard
+  drawDebounce: any
+}
+
+class Canvas extends React.Component<CanvasProps, CanvasState> {
+  constructor(props){
+    super(props)
+    this.state = {
+      card: new CanvasCard(),
+      drawDebounce: undefined
+    }
+  }
+
+  componentDidMount(){
+    this.setState({drawDebounce : debounce(this.draw, 200)});
+  }
+
+  shouldComponentUpdate(nextProps){
+    return this.props.furniture != nextProps.furniture;
+  }
+
+  componentDidUpdate() {
+    const canvas = this.refs.canvas as HTMLCanvasElement;
+    this.state.drawDebounce.clear();
+    this.state.drawDebounce(canvas, this.state, this.props)
+
+  }
+
+  draw(canvas: HTMLCanvasElement, state: CanvasState, props: CanvasProps) {
+    state.card.draw(canvas, props.furniture)
+      .then(() => canvas.toBlob( (blob) => props.update(blob)));
+  }
+
+  render() {
+    return (
+      <canvas ref="canvas" width="500px" height="500px" className="card"></canvas>
```

Comment:
Should the canvas dimensions be derived from the device choice?

Reply:
Well spotted, I think they're redundant because the width and height gets set when rendering the canvas. I will remove them.